text/006-local-js-dependencies.md

```diff
@@ -0,0 +1,347 @@
+now this should be sufficient for expressiveness.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* The initial RFC is fairly conservative. It doesn't work with `--nodejs` out of
+  the gate nor `--no-modules`. Additionally it doesn't support JS snippets
+  importing other JS initially. Note that all of these are intended to be
+  supported in the future, it's just thought that it may take more design than
+  we need at the get-go for now.
+
+* JS snippets must be written in vanilla ES module JS syntax. Common
+  preprocessors like TypeScript can't be used. It's unclear how such
+  preprocessed JS would be imported. It's hoped that JS snippets are small
+  enough that this isn't too much of a problem. Larger JS snippets can always be
+  extracted to an NPM package and postprocessed there. Note that it's always
+  possible for authors to manually run the TypeScript compiler by hand for these
+  use cases, though.
+
+* The relatively popular `--no-modules` flag is proposed to be deprecated in
+  favor of a `--browser` flag, which itself will have a breaking change relative
+  to today. It's thought though that `--browser` is only very rarely used so is
+  safe to break, and it's also thought that we'll want to avoid breaking
+  `--no-modules` as-is today.
+
+* JS files are imported relative to the crate root rather than the file doing
+  the importing, unlike `mod` statements in Rust. It's thought that we can't get
+  file-relative imports working with stable `proc_macro` APIs today, but if the
+  implementation can manage to finesse it this RFC would propose instead making
+  file imports relative to the current file instead of crate root.
+
+* Local JS snippets are required to be written in ES module syntax. This may be
+  a somewhat opinionated stance, but it's intended to make it easier to add
+  future features to `wasm-bindgen` while continuing to work with JS.
+
+# Rationale and Alternatives
+[alternatives]: #rationale-and-alternatives
+
+Currently there aren't any competing designs solving this problem, so there
+aren't many known major alternatives. There's a number of alternatives to
+various smaller decisions in this RFC, but it's hoped that the various
+alternatives are listed or discussed inlined.
+
+The major rationale for this RFC is empowering this use case of "seamless local
+JS snippets" **at all**. The RFC proposes to start out with only including files
+as opposed to small JS snippets themselves (like `stdweb`'s own `js!` macro)
+because files are structured as ES modules which is what `#[wasm_bindgen]`
+already supports well and will also eventually have support natively in
+WebAssembly itself. While `#[wasm_bindgen]` may one day have a `js!`-like macro
+built-in, it's hoped that we can start out with a purely library-based solution
+like `stdweb`, iterate on it, and consider this question later.
+
+One alternative for ES modules is to simply concatenate all JS together. This
```

Comment:
Since we're requiring ES6 modules anyways, "forcing everyone to agree on the module format" isn't a problem.

Name conflicts are indeed an issue, so ideally we'd want to use an already-existing ES6 module concatenator (which do exist). So that way we avoid reinventing the wheel.